update HF-Inference to inference providers only (#1809)
huggingface/hub-docs: docs/hub/models-inference.md

## Build with Inference Providers

You can integrate Inference Providers into your own applications using our SDKs or HTTP clients. Here's a quick start with Python and JavaScript, for more details, check out the [Inference Providers Documentation](https://huggingface.co/docs/inference-providers).

<hfoptions id="inference-providers-quick-start">

<hfoption id="python">

You can use our Python SDK to interact with Inference Providers.

```python
from huggingface_hub import InferenceClient

import os

client = InferenceClient(
    api_key=os.environ["HF_TOKEN"],
    provider="auto",   # Automatically selects best provider
)

# Chat completion
completion = client.chat.completions.create(
    model="deepseek-ai/DeepSeek-V3-0324",
    messages=[{"role": "user", "content": "A story about hiking in the mountains"}]
)

# Image generation
image = client.text_to_image(
    prompt="A serene lake surrounded by mountains at sunset, photorealistic style",
    model="black-forest-labs/FLUX.1-dev"
)

```

Or, you can just use the OpenAI API compatible client.

> ```python

Reviewer: was this supposed to be OAI client?

# Inference Providers

Hugging Face's model pages have free inference for thousands of models, so you can try them all out right in the browser. It's also powered by Inference Providers.

Inference Providers give developers streamlined, unified access to hundreds of machine learning models, powered by the best serverless inference partners. 👉 **For complete documentation, visit the [Inference Providers Documentation](https://huggingface.co/docs/inference-providers)**.

## Inference Providers on the Hub

Inference Providers is deeply integrated with the Hugging Face Hub, and you can use it in a few different ways:

> - **Interactive Widgets** - Test models directly on model pages with interactive widgets that use Inference Providers under the hood. Check out the [DeepSeek-R1-0528 model page](https://huggingface.co/models/deepseek-ai/DeepSeek-R1-0528) for an example.

Reviewer 2: The model URL is invalid here

```suggestion
- **Interactive Widgets** - Test models directly on model pages with interactive widgets that use Inference Providers under the hood. Check out the [DeepSeek-R1-0528 model page](https://huggingface.co/deepseek-ai/DeepSeek-R1-0528) for an example.
```

- **Inference Playground** - Easily test and compare chat completion models with your prompts. Check out the [Inference Playground](https://huggingface.co/playground) to get started.
- **Search** - Filter models by inference provider on the [models page](https://huggingface.co/models?inference_provider=all) to find models available through specific providers.

## Build with Inference Providers

You can integrate Inference Providers into your own applications using our SDKs or HTTP clients. Here's a quick start with Python and JavaScript, for more details, check out the [Inference Providers Documentation](https://huggingface.co/docs/inference-providers).

<hfoptions id="inference-providers-quick-start">

<hfoption id="python">

```python
from huggingface_hub import InferenceClient

> client = InferenceClient(provider="auto")  # Automatically selects best provider

Reviewer 2: ```suggestion
import os

client = InferenceClient(
    api_key=os.environ["HF_TOKEN"],
    provider="auto",   # Automatically selects best provider
)
```

# Inference Providers

> Hugging Face's model pages have free inference for thousands of models, so you can try them all out right in the browser. It's also powered by Inference Providers.

Reviewer: not free, it's paid

Reviewer 3: ```suggestion
Hugging Face's model pages have pay-as-you-go inference for thousands of models, so you can try them all out right in the browser. Service is powered by Inference Providers and includes a free-tier.
```

(maybe like this to mention it's paid with a free-tier?)

Inference Providers give developers streamlined, unified access to hundreds of machine learning models, powered by the best serverless inference partners. 👉 **For complete documentation, visit the [Inference Providers Documentation](https://huggingface.co/docs/inference-providers)**.

## Inference Providers on the Hub

Inference Providers is deeply integrated with the Hugging Face Hub, and you can use it in a few different ways:

- **Interactive Widgets** - Test models directly on model pages with interactive widgets that use Inference Providers under the hood. Check out the [DeepSeek-R1-0528 model page](https://huggingface.co/deepseek-ai/DeepSeek-R1-0528) for an example.
- **Inference Playground** - Easily test and compare chat completion models with your prompts. Check out the [Inference Playground](https://huggingface.co/playground) to get started.
- **Search** - Filter models by inference provider on the [models page](https://huggingface.co/models?inference_provider=all) to find models available through specific providers.

> 

Reviewer 3: (nit) Data Studio is also powered by Inference Providers as far as I know: https://huggingface.co/datasets/fka/awesome-chatgpt-prompts/viewer?views%5B%5D=train

## Build with Inference Providers

You can integrate Inference Providers into your own applications using our SDKs or HTTP clients. Here's a quick start with Python and JavaScript, for more details, check out the [Inference Providers Documentation](https://huggingface.co/docs/inference-providers).

<hfoptions id="inference-providers-quick-start">

<hfoption id="python">

```python

> from huggingface_hub import InferenceClient

Reviewer: make these complete w/ huggingface_hub login

Reviewer 3: I would keep "as is" to be honest. The snippet already includes `api_key=os.environ["HF_TOKEN"],` (which by the way is not even needed but good to let the users know about the env variable). This is the code snippet we generate on model pages.

Reviewer 3: (I just realized we should go for `os.getenv("HF_TOKEN")` instead of `os.environ["HF_TOKEN"]`... First one will work if token is saved in file instead of env variable while the second will fail even though the user is logged in. I'll make a pass to fix that everywhere :disappointed:

import os

client = InferenceClient(
    api_key=os.environ["HF_TOKEN"],
    provider="auto",   # Automatically selects best provider
)

# Chat completion
completion = client.chat.completions.create(
    model="deepseek-ai/DeepSeek-V3-0324",
    messages=[{"role": "user", "content": "Hello!"}]
)
```

</hfoption>

<hfoption id="javascript">

> ```javascript

Reviewer: we can use another Text to Image example here

import { InferenceClient } from "@huggingface/inference";

const client = new InferenceClient(process.env.HF_TOKEN);

const chatCompletion = await client.chatCompletion({
    provider: "auto",  // Automatically selects best provider  
    model: "deepseek-ai/DeepSeek-V3-0324",
    messages: [{ role: "user", content: "Hello!" }]
});
```

</hfoption>

</hfoptions>

> 

Reviewer: Let's add a OAI compat snippet here

You'll need a Hugging Face token with inference permissions. Create one at [Settings > Tokens](https://huggingface.co/settings/tokens/new?ownUserPermissions=inference.serverless.write&tokenType=fineGrained).

### How Inference Providers works

> Hugging Face’s Inference Providers give developers unified access to hundreds of machine learning models, powered by our serverless inference partners. This new approach builds on our previous Serverless Inference API, offering more models, improved performance, and greater reliability thanks to world-class providers.

Reviewer: Oof, let's rewrite this, it sounds too LLM-y

Reviewer 3: IMO we can just remove this paragraph and keep the section

```md
### How Inference Providers works

To dive deeper into Inference Providers, check out the [Inference Providers Documentation](https://huggingface.co/docs/inference-providers). Here are some key resources:
...
```

We already explain what Inference Providers is above in the page.

Author: I'll drop it.

@Vaibhavs10 I lift it from here: https://huggingface.co/docs/inference-providers/index where it's the first paragraph. so maybe we should revise in https://github.com/huggingface/hub-docs/pull/1806
